kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -454,11 +454,8 @@ object Abort:
       */
     def catching[E](
         using Frame
-    )[A, S](v: => A < S)(using ct: SafeClassTag[E]): A < (Abort[E] & S) =
-        Effect.catching(v) {
-            case ct(ex) => Abort.fail(ex)
-            case ex     => Abort.panic(ex)
-        }
+    )[A, S](v: => A < S): A < (Abort[E] & S) =
+        Effect.defer(v)
```

Comment:
This method and `Effect.catching` were designed to enable error handling without requiring a `Flat` evidence. Given that we don't have `Flat` anymore, I agree we can remove `Effect.catching` and migrate all error handling to `Abort.runWith`. 

`Effect.defer` here won't catch anything, though. I imagine we're missing tests for this method since the build isn't failing. This can use `Abort.runWith` instead.